### ISSUE TITLE
Component: Tristatecheckbox

### ISSUE THREAD

**Issue by sergio-carlavilla** [primefaces/primeng#17137]
### Describe the bug

What happened with the Tristatecheckbox component in PrimeNG19?
Seems it is not available anymore. Is a bug?

### Environment

Angular 19 with PrimeNG 19

### Angular version

19.0.0

### PrimeNG version

19.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.14.0

**Comment by mastersys-giovanni** [comment]
This component no longer exists. Use p-checkbox with binary=false

**Comment by sergio-carlavilla** [comment]
Visually it is not the same, for a user used to seeing the 3 options, it is not the same.

**Comment by mastersys-giovanni** [comment]
<p-checkbox [(ngModel)]="checked" [binary]="true" [indeterminate]="true" />

**Comment by sergio-carlavilla** [comment]
Yes, I know about this option, but it's not the same.

**Comment by cetincakiroglu** [comment]
Hi,

It's removed and it's announced in migration guide https://primeng.org/guides/migration

Closing the issue, you can use indeterminate checkbox instead https://primeng.org/guides/migration

**Comment by adama357** [comment]
@cagataycivici @cetincakiroglu It doesn't seem like TriStateCheckbox should've been removed altogether in v18, since it wasn't deprecated in v17.

Also, I created a discussion [here](https://github.com/orgs/primefaces/discussions/3683) about the fact that the Checkbox indeterminate option is not a sufficient substitute for the TriStateCheckbox:

 
tl;dr:

1. The whole point of the TriStateCheckbox is that you can cycle **back to a null value.** Indeterminate Checkboxes don't do that; they're just able to indicate a null state _at first_, before you click on them. TriStateCheckboxes and indeterminate Checkboxes aren't the same thing.

2. If you're going to merge the functionally of TriStateCheckbox into one of the other components, the ToggleButton would be a better choice imho. ToggleButtons have labels, and a TriStateCheckbox without a label would be very confusing. 

Thank you for your time. I know there are a lot of PrimeNG issue tickets... ^^
